feat: sqlite implementation (#438)
googleapis/genai-toolbox: docs/en/resources/tools/sqlite-sql.md

---
title: "SQLite SQL"

> linkTitle: "SQLite SQL"

Reviewer: ```suggestion
```

type: docs
weight: 1
description: >
  Execute SQL statements against a SQLite database.
---

## About

A `sqlite-sql` tool executes SQL statements against a SQLite database.
It's compatible with any of the following sources:

- [sqlite](../sources/sqlite.md)

> ## Parameters
> 
> SQLite uses the `?` placeholder for parameters in SQL statements. Parameters are

Reviewer: ```suggestion
SQLite uses the `?` placeholder for parameters in SQL statements. Parameters are
```
To keep consistency with the other docs.

bound in the order they are provided.

> 

Reviewer: ```suggestion

The statement field supports any valid SQLite SQL statement, including `SELECT`, `INSERT`, `UPDATE`, `DELETE`, `CREATE/ALTER/DROP` table statements, and other DDL statements.

```

### Example

```yaml

> tools:
>   get-user:
>     kind: sqlite-sql
>     source: my-sqlite-db
>     description: Get user information by ID
>     parameters:
>       - name: id
>         type: integer
>         description: The user ID to look up
>     statement: SELECT * FROM users WHERE id = ?
> ```
> 
> Multiple parameters:
> ```yaml

Reviewer: ```suggestion
```
One example is sufficient :)

tools:
  search-users:
    kind: sqlite-sql
    source: my-sqlite-db
    description: Search users by name and age
    parameters:
      - name: name
        type: string
        description: The name to search for
      - name: min_age
        type: integer
        description: Minimum age
    statement: SELECT * FROM users WHERE name LIKE ? AND age >= ?
```

> ## Reference
> 
> ### Configuration Options

Reviewer: ```suggestion
## Reference
```

> | Field | Type | Required | Description |
> |-------|------|----------|-------------|

Reviewer: ```suggestion
| **field**   |                  **type**                  | **required** | **description**                                                                                  |
|-------------|:------------------------------------------:|:------------:|--------------------------------------------------------------------------------------------------|

```

> | kind | string | Yes | Must be "sqlite-sql" |
> | source | string | Yes | Name of a SQLite source configuration |
> | description | string | Yes | Description of what the tool does |
> | parameters | array | No | List of parameters for the SQL statement |
> | statement | string | Yes | The SQL statement to execute |

Reviewer: ```suggestion
| kind        |                   string                   |     true     | Must be "sqlite-sql".                                                                          |
| source      |                   string                   |     true     | Name of the source the SQL statement should execute on.                                                    |
| description |                   string                   |     true     | Description of the tool that is passed to the LLM.                                               |
| statement   |                   string                   |     true     | SQL statement to execute on.                                                                     |
| parameters  | [parameters](_index#specifying-parameters) |    false     | List of [parameters](_index#specifying-parameters) that will be inserted into the SQL statement. |
```

> 
> ### Parameter Types
> 
> The following parameter types are supported:
> - `string`
> - `integer`
> - `float`
> - `boolean`
> - `array`
> - `object`
> 
> ### SQL Statement
> 
> The statement field supports any valid SQLite SQL statement, including:
> - SELECT queries
> - INSERT statements
> - UPDATE statements
> - DELETE statements
> - CREATE/ALTER/DROP table statements
> - Other DDL statements

Reviewer: ```suggestion
```

User will have to use the parameter types defined in Toolbox. For Type of SQL statement, I moved it to the `About` section.
